pkg/component/component.go

```diff
@@ -239,3 +243,83 @@ func Exists(client *occlient.Client, componentName, applicationName, projectName
 	}
 	return false, nil
 }
+
+// LinkInfo contains the information about the link being created between
+// components
+type LinkInfo struct {
+	SourceComponent string
+	TargetComponent string
+	Envs            []string
+}
+
+// Link injects connection information of the target component into the source
+// component as environment variables
+func Link(client *occlient.Client, sourceComponent, targetComponent, applicationName string) (*LinkInfo, error) {
+	// Get Service associated with the target component
+	targetService, err := client.GetOneServiceFromSelector(
```

Comment:
my head hurts from trying to parse this :-( 
Would it be better to have separate this? It is hard to parse 3 nested function calls :(

Reply:
haha, fixed